server/src/main/java/io/crate/auth/HostBasedAuthentication.java

```diff
@@ -172,6 +173,7 @@ Optional<Map.Entry<String, Map<String, String>>> getEntry(@Nullable String user,
             .filter(e -> Matchers.isValidAddress(e.getValue().get(KEY_ADDRESS), connectionProperties.address(), dnsResolver))
             .filter(e -> Matchers.isValidProtocol(e.getValue().get(KEY_PROTOCOL), connectionProperties.protocol()))
             .filter(e -> Matchers.isValidConnection(e.getValue().get(SSL.KEY), connectionProperties))
+            .filter(e -> Matchers.isValidMethod(e.getValue().get(KEY_METHOD), connectionProperties.clientMethods()))
             .findFirst();
```

Comment:
It's okay to do `findFirst` since on each entry we check ALL client methods.
Added `test_multiple_matches_after_properties_check_pick_up_first` to cover a case we talked about.